VSTS CI Mac Tests hangs

The following test hangs:
test_exception_breakpoints (tests.highlevel.test_live_pydevd.BreakpointTests) ...

For some reason the tests running on Mac hang (randomly) at the same place, even though the test is disabled:

```
2018-07-09T17:07:01.9310680Z   test_attach_exit_during_session (tests.highlevel.test_lifecycle.LifecycleTests) ... SKIP (0.000s)
2018-07-09T17:07:04.0875620Z   test_attach_from_unix_os (tests.highlevel.test_lifecycle.LifecycleTests) ... OK (2.160s)
2018-07-09T17:07:05.9403340Z   test_attach_from_windows_os (tests.highlevel.test_lifecycle.LifecycleTests) ... OK (2.172s)
2018-07-09T17:07:05.9418780Z   test_launch (tests.highlevel.test_lifecycle.LifecycleTests) ... OK (0.024s)
2018-07-09T17:07:06.2616890Z   test_breakpoints_single_file (tests.highlevel.test_live_pydevd.BreakpointTests) ... OK (0.321s)
2018-07-09T17:36:51.8228630Z ##[debug]Re-evaluate condition on job cancellation for step: 'Run Tests'.
2018-07-09T17:36:51.8246340Z ##[debug]Skip Re-evaluate condition on agent shutdown.
2018-07-09T17:36:51.8438370Z   test_exception_breakpoints (tests.highlevel.test_live_pydevd.BreakpointTests) ... 
2018-07-09T17:36:52.0201090Z ##[error]The operation was canceled.
2018-07-09T17:36:52.0271590Z ##[debug]System.OperationCanceledException: The operation was canceled.
   at System.Threading.CancellationToken.ThrowOperationCanceledException()
```

https://ptvsd.visualstudio.com/ptvsd/ptvsd%20Team/_build/results?buildId=337&view=logs

## Notes:
* The test has been disabled, yet it hangs just when attempting to run that test.
* We must add timeouts for these tests, as done in the system tests.

/cc @ericsnowcurrently @karthiknadig @d3r3kk 
Thoughts?

--- a/tests/highlevel/__init__.py
+++ b/tests/highlevel/__init__.py
@@ -919,6 +919,12 @@ class VSCTest(object):
         expected = list(daemon.protocol.parse_each(expected))
         self.assertEqual(received, expected)
 
+    def assert_contains(self, received, expected):
+        from tests.helpers.message import assert_contains_messages
+        received = list(self.vsc.protocol.parse_each(received))
+        expected = list(self.vsc.protocol.parse_each(expected))
+        assert_contains_messages(received, expected)
+
     def assert_received_unordered_payload(self, daemon, expected):
         """Ensure that the received messages match the expected ones
         regardless of payload order."""

--- a/tests/highlevel/test_live_pydevd.py
+++ b/tests/highlevel/test_live_pydevd.py
@@ -4,6 +4,7 @@ import contextlib
 import os
 import sys
 from textwrap import dedent
+import time
 import traceback
 import unittest
 
@@ -304,14 +305,12 @@ class BreakpointTests(VSCFlowTest, unittest.TestCase):
         with captured_stdio() as (stdout, _):
             with self.launched(config=config):
                 # Allow the script to run to completion.
-                received = self.vsc.received
+                time.sleep(1.)
         out = stdout.getvalue()
 
         for req, _ in self.lifecycle.requests:
             self.assertNotEqual(req['command'], 'setBreakpoints')
             self.assertNotEqual(req['command'], 'setExceptionBreakpoints')
-        self.assert_received(self.vsc, [])
-        self.assert_vsc_received(received, [])
         self.assertIn('2 4 4', out)
         self.assertIn('ka-boom', out)
 
@@ -403,7 +402,6 @@ class BreakpointTests(VSCFlowTest, unittest.TestCase):
         self.assertIn('2 4 4', out)
         self.assertIn('ka-boom', err)
 
-    @unittest.skip('not working right #614')
     def test_exception_breakpoints(self):
         self.vsc.PRINT_RECEIVED_MESSAGES = True
         done, script = self._set_lock('h')
@@ -422,7 +420,9 @@ class BreakpointTests(VSCFlowTest, unittest.TestCase):
                     _, tid = self.get_threads(self.thread.name)
                 with self.wait_for_event('stopped'):
                     done()
-
+                req_continue_last = self.send_request('continue', {
+                    'threadId': tid,
+                })
                 # Allow the script to run to completion.
                 received = self.vsc.received
         out = stdout.getvalue()
@@ -445,6 +445,8 @@ class BreakpointTests(VSCFlowTest, unittest.TestCase):
                  threadId=tid,
                  text='MyError',
                  description=description)),
+            req_continue_last,
+            ('continued', dict(threadId=tid, )),
         ])
         self.assertIn('2 4 4', out)
         self.assertIn('ka-boom', out)

--- a/tests/highlevel/test_live_pydevd.py
+++ b/tests/highlevel/test_live_pydevd.py
@@ -427,7 +427,6 @@ class BreakpointTests(VSCFlowTest, unittest.TestCase):
                 # Allow the script to run to completion.
                 received = self.vsc.received
         out = stdout.getvalue()
-        #print(' + ' + '\n + '.join(out.splitlines()))
 
         got = []
         for req, resp in self.lifecycle.requests:
@@ -439,15 +438,14 @@ class BreakpointTests(VSCFlowTest, unittest.TestCase):
             description = "MyError('ka-boom')"
         else:
             description = "MyError('ka-boom',)"
-        self.assert_vsc_received_fixing_events(received, [
-            ('stopped',
-             dict(
+        self.assert_contains(received, [
+            self.new_event('stopped', **dict(
                  reason='exception',
                  threadId=tid,
                  text='MyError',
                  description=description)),
-            req_continue_last,
-            ('continued', dict(threadId=tid, )),
+            self.new_response(req_continue_last),
+            self.new_event('continued', **dict(threadId=tid, )),
         ])
         self.assertIn('2 4 4', out)
         self.assertIn('ka-boom', out)

--- a/tests/highlevel/test_live_pydevd.py
+++ b/tests/highlevel/test_live_pydevd.py
@@ -420,9 +420,10 @@ class BreakpointTests(VSCFlowTest, unittest.TestCase):
                     _, tid = self.get_threads(self.thread.name)
                 with self.wait_for_event('stopped'):
                     done()
-                req_continue_last = self.send_request('continue', {
-                    'threadId': tid,
-                })
+                with self.wait_for_event('continued'):
+                    req_continue_last = self.send_request('continue', {
+                        'threadId': tid,
+                    })
                 # Allow the script to run to completion.
                 received = self.vsc.received
         out = stdout.getvalue()

